### DISCUSSION TITLE
classes/class_classdb

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#111] [User-contributed notes]
# classes/class_classdb

Inherits: Object A class information repository. Description: Provides access to metadata stored for every available class. Methods:,, bool, can_instantiate(class: StringName) const,, bool, class_e...

https://docs.godotengine.org/en/stable/classes/class_classdb.html

**Comment by RoughEdgeBarb** [comment]
get_class_list() only returns the engine classes, if you are looking for loaded scripts then use ProjectSettings.get_global_class_list()

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot/pull/109747

**Comment by Varonex** [comment]
I think it should be mentioned somewhere that "every available class" only refers to engine class, not all classes project-wise. That means classes defined via class_name will not return any reflection info when using `class_get_method_list`, `class_get_property_list`, and so on.

**Reply by Calinou** [reply]
Pull request opened:

- https://github.com/godotengine/godot/pull/109747
